packages/react-native-bridge/index.js

```diff
@@ -529,4 +534,60 @@ export function toggleRedoButton( isDisabled ) {
 	RNReactNativeGutenbergBridge.toggleRedoButton( isDisabled );
 }
 
+/**
+ * Log exception to host app's crash logging service.
+ * @param {Object}   exception         Exception object
+ * @param {Object}   [extra]           Extra parameters to include in the exception.
+ * @param {Object}   [extra.context]   Context of the exception.
+ * @param {Object}   [extra.tags]      Tags to associate with the exception.
+ * @param {Object}   [extra.isHandled] True if the exception is handled.
+ * @param {Object}   [extra.handledBy] The mechanism that detected the exception.
+ * @param {Function} [callback]        Callback triggered when the exception is sent.
+ */
+export function logException(
+	exception,
+	{ context, tags, isHandled, handledBy } = {
+		context: {},
+		tags: {},
+		isHandled: false,
+		handledBy: 'Unknown',
+	},
+	callback
+) {
+	const parsedException = {
+		...parseException( exception, { context, tags } ),
+		isHandled,
+		handledBy,
+	};
+
+	const onLogException = ( wasSent ) => {
+		if ( ! wasSent ) {
+			// eslint-disable-next-line no-console
+			console.error(
+				'An error ocurred when logging the exception',
+				parsedException
+			);
+		}
+		if ( callback ) {
+			callback();
+		}
+	};
+
+	// Only log exceptions in production
+	// eslint-disable-next-line no-undef
+	if ( __DEV__ ) {
+		// eslint-disable-next-line no-console
+		console.info( 'Exception that would be logged', parsedException );
+		if ( callback ) {
+			callback();
+		}
+		return;
+	}
+
+	RNReactNativeGutenbergBridge.logException(
+		parsedException,
+		onLogException
+	);
```

Comment:
Is there any difference if `onLogException` is called right after `logException`, instead of a callback?

```javascript
RNReactNativeGutenbergBridge.logException(parsedException);
onLogException();
```

Reply:
Yes, all React Native bridge functions are asynchronous ([reference](https://reactnative.dev/docs/0.71/native-modules-ios#callbacks)) by default. So calling `onLogException` after `logException` won't guarantee that the exception is sent when calling `onLogException`.